text/0000-fabric-x-next-generation.md

```diff
@@ -0,0 +1,365 @@
+    * The introduction of support for traditional chaincode, refined PDCs, and multiple channels would be subject to separate, future RFCs once the new governance model for coexisting Fabric implementations is established and the core Fabric-X platform has stabilized.
+* **Coexistence and Targeted Adoption:**
+    * It is crucial to understand that **migration to Fabric-X is not intended or necessary for all existing Fabric networks.** Fabric (Classic) is a mature, stable platform that effectively serves a vast number of production use cases.
+    * Fabric-X is primarily aimed at **new, demanding use cases** that require performance levels significantly beyond what current Fabric can offer (e.g., CBDCs, high-frequency digital asset exchanges).
+    * Organizations should only consider Fabric-X if their performance requirements genuinely necessitate it. Otherwise, they would face the **unnecessary operational complexity** of a distributed microservice deployment and a potentially involved migration process without fully benefiting from its core advantage—extreme performance. Most existing Fabric use cases are well-served by its current performance capabilities.
+    * The vision is for Fabric and Fabric-X to **coexist within the broader Hyperledger Fabric ecosystem**, each serving different segments and requirements. Fabric-X expands the ecosystem's capabilities, rather than seeking to universally replace the established Fabric implementation.
+
+The primary aim of Fabric-X is to enable new, high-performance use cases and ensure the Fabric ecosystem remains at the forefront of innovation for next-generation DLT applications.
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+This section details the technical design of Fabric-X. Fabric-X implements the execute-order-validate flow using the following microservice architecture.
+
+## 1. Microservice Architecture and Distributed Database (Peer-Side)
+
+The monolithic Fabric Peer is decomposed into the independently scalable microservices as shown in the below figure.
+![Proposed Fabric Peer Architecture](../images/fabx/peer-arch.png)
+* **Endorser Service:** Receives transaction proposals from clients. It executes the specified application endorsement logic (e.g., as part of a 'Flow' implemented with FSC or using an endorser SDK) against a snapshot read from the organization's sharded distributed database through a query service and generates a read-write set along with its signature. Multiple instances can be deployed to scale the endorsement capacity. Each organization's endorser will also maintain a local database for application-specific needs (e.g., managing off-chain state for Flows, or sensitive data not intended for the main ledger).
```

Comment:
> distributed database

What is the fault tolerance model? Is it BFT or CFT?

Reply:
Each organization hosts a distributed database similar to how CouchDB is hosted currently.